packages/vite-plugin/src/main.ts

```diff
@@ -121,6 +138,11 @@ export default ({
     async configResolved(resolvedConfig) {
       viteConfig = resolvedConfig;
     },
+    resolveId(id) {
+      if (id === proxySettingsKey) {
+        return '\0' + proxySettingsKey;
```

Comment:
Extract this and the one below out into a constant, like `resolvedEditorId` (build it from proxySettingsKey though, you don't have to repeat yourself)